### ISSUE TITLE
PropertyGrid - Searchbox support for finding child properties (within ExpandableObjects)

### ISSUE THREAD

**Issue by kcris** [xceedsoftware/wpftoolkit#1691]
Hi 

@XceedBoucherS 
PropertyGrid allows displaying expandable objects with their own childs.
But the SearchBox only finds top-level properties (tried v3.8.2.0 and 4.0.0.0).
Is there a way to scan all properties and find any child, at any depth?
I could not find such an option in the docs

Thanks

**Comment by XceedBoucherS** [comment]
Hello,
Sorry for the long delay,
I just tested Toolkit v4.0.0 from NuGet for a PropertyGrid with some properties containing subProperties. The searchbox returned the expected searched properties, even the ones from sub-properties. Upon starting the sample, I type in the searchBox a sub-property name, like "street". Here's my sample:

`<Grid>
    <xctk:PropertyGrid SelectedObject="{Binding}"/>
  </Grid>`

`  public partial class MainWindow : Window
  {
    public MainWindow()
    {
      InitializeComponent();

      this.DataContext = new Player()
      {
        Info = new PlayerInfo() { FirstName = "Carey", LastName = "Price", Address = new Address() { City = "Mtl", Street = "Main" } },
        Age = 34
      };
    }
  }

  public class Player
  {
    [ExpandableObject()]
    public PlayerInfo Info
    {
      get;
      set;
    }

    public int Age
    {
      get;
      set;
    }
  }

  public class PlayerInfo
  {
    public string FirstName
    {
      get;
      set;
    }

    public string LastName
    {
      get;
      set;
    }

    [ExpandableObject()]
    public Address Address
    {
      get;
      set;
    }
  }

  public class Address
  {
    public string Street
    {
      get;
      set;
    }

    public string City
    {
      get;
      set;
    }
  }`

**Comment by XceedBoucherS** [comment]
Hi,
Can you submit a sample showing the issue.
I'm curious to see the differences between your version and my test.
Thank you.

**Comment by kcris** [comment]
Hi, 

here is a sample
[WpfApp2.zip](https://github.com/xceedsoftware/wpftoolkit/files/7695745/WpfApp2.zip)

![image](https://user-images.githubusercontent.com/812733/145648884-a8f581bf-4b61-484e-bfa0-31c9ed8084df.png)

![image](https://user-images.githubusercontent.com/812733/145648900-3e0ff81a-ce03-4294-b60c-27bf56d137b1.png)

Thank you @XceedBoucherS

**Comment by kcris** [comment]
actually, I used the Extended version for the sample, that was a mistake. 
What I really need is the free/opensource version. Can you run the test with that version? Thanks

**Comment by XceedBoucherS** [comment]
Hi,

I've created a new .NET Framework app (not .NET5), so I could use the OpenSource "Extended.Wpf.Toolkit" version. I downloaded it from NuGet : v4.2.
I then updated the MainWindows.xaml and MainWindow.xaml.cs and added your ViewModel.cs.

The result is that sub-properties are not displayed when searching with the searchBox, like you found.

Conclusion : search of sub-properties are only part of the Plus version. You will need to the Plus version in order to have this feature. The openSource version search only for top level properties.
